app/src/main/java/com/codepath/apps/critterfinder/FindActivity.java

```diff
@@ -20,14 +22,18 @@
 import java.util.ArrayList;
 
 import cz.msebera.android.httpclient.Header;
+import permissions.dispatcher.NeedsPermission;
+import permissions.dispatcher.RuntimePermissions;
 
 
-public class FindActivity extends AppCompatActivity {
+@RuntimePermissions
```

Comment:
annotation library for dealing with permissions - https://github.com/hotchemi/PermissionsDispatcher